src/types.ts

```diff
@@ -381,10 +381,18 @@ export type ExportUsersResponse = {
   task_id: string;
 };
 
+export type ExportChannelsResult = {
+  path?: string;
+  s3_bucket_name?: string;
+  url?: string;
+};
+
 export type ExportChannelStatusResponse = {
+  status: string;
+  task_id: string;
   created_at?: string;
-  error?: {};
-  result?: {};
+  error?: ExportChannelsResult | null;
+  result?: {} | null;
```

Comment:
`result` should use `ExportChannelsResult`

and `error` should use `ErrorResult`

where `ErrorResult` is

```
type?: string;
description?: string;
stacktrace?: string;
version?: string;
```

Reply:
Thanks. That was embarrassing :)

Applied your suggestion with the difference that I've called the error type `ExportChannelError` since I thought `ErrorResult` would be too generic. Can change if you actually want it to be that generic.

Reply:
Thanks, also I think `null` is unnecessary here . `ExportChannelError` sounds good to me 👍

Reply:
and we would need to make it

```
 status?: string;
 task_id?: string;
```

Reply:
Don't fully understand. Add those to the `ExportChannelError` or change required to optional in `ExportChannelStatusResponse`?

> and we would need to make it
> 
> ```
>  status?: string;
>  task_id?: string;
> ```

Reply:
Sorry, to be lear, what I meant is that all the properties in `ExportChannelStatusResponse` must be made optional, like below

```
export type ExportChannelStatusResponse = {
  status?: string;
  task_id?: string;
  created_at?: string;
  error?: ExportChannelError;
  result?: ExportChannelsResult | null;
  updated_at?: string;
};
```

Reply:
can you also remove the null here please 

```
result?: ExportChannelsResult | null;
```

Reply:
```
Status {
  task_id: 'c1e57bb1-04f1-4284-af61-455c3990b631',
  status: 'waiting',
  created_at: '2023-11-30T12:42:46.266232Z',
  updated_at: '2023-11-30T12:42:46.266232Z',
  duration: '16.19ms',
  result: null
}
Status {
  task_id: 'c1e57bb1-04f1-4284-af61-455c3990b631',
  status: 'pending',
  created_at: '2023-11-30T12:42:46.266232Z',
  updated_at: '2023-11-30T12:42:46.501436Z',
  duration: '3.45ms',
  result: null
}
Status {
  task_id: 'c1e57bb1-04f1-4284-af61-455c3990b631',
  status: 'completed',
  created_at: '2023-11-30T12:42:46.266232Z',
  updated_at: '2023-11-30T12:42:46.689775Z',
  duration: '2.00ms',
  result: {
    url: 'https://...'
  }
}
```

Reply:
BTW. I wonder what makes you think that `task_id` and `status` are optional. Did you verify internally that it can be the case that those are missing? I would think that the response would be pretty useless without those.

Reply:
We're closing on a year here.
If you don't care then I can just close it.